src/lib/device/device.h

```diff
@@ -159,12 +159,16 @@ class DeviceManager {
 
     const Chunk* get_chunk(uint32_t chunk_id) const {
         std::unique_lock lg{m_vdev_mutex};
-        return (chunk_id == INVALID_CHUNK_ID) ? nullptr : m_chunks[chunk_id].get();
+        // if a pdev is misssing when restart, chunk_id from cleint might be larger than m_chunks.size()
+        if (chunk_id == INVALID_CHUNK_ID || chunk_id >= m_chunks.size()) return nullptr;
+        return m_chunks[chunk_id].get();
```

Comment:
can we just `return get_chunk_mutable(chunk_id);`" for better readability ?

Reply:
we can not call `get_chunk_mutable(chunk_id)` inside `get_chunk` since `get_chunk` is a `const func() const` but `get_chunk_mutable` is non-const